File: 프로그래머스/lv2.위장.py
# Level2: 위장
from collections import defaultdict
def solution(clothes):
    answer = 1
    closet = defaultdict(list)
    for c,t in clothes:
        closet[t].append(c)
    for c in closet:
        answer *= (len(closet[c])+1)
    return answer-1

File: 프로그래머스/lv2.위장.js
// Level2: 위장
function solution(clothes) {
    let answer = 1;
    let closet = new Map()
    clothes.forEach((ele) => {
        if(closet.has(ele[1])) {
            let arr = closet.get(ele[1]);
            arr.push(ele[0]);
            closet.set(ele[1],arr);
        }
        else closet.set(ele[1],[ele[0]]);
    });
    closet.forEach((ele) => {
        answer *= (ele.length+1);
    });
    return answer-1;
}